node/src/index.ts

```diff
@@ -44,31 +44,62 @@ export async function connect (uri: string): Promise<Connection> {
 export interface Connection {
   uri: string
 
-  tableNames: () => Promise<string[]>
+  /* eslint-disable @typescript-eslint/method-signature-style */
```

Comment:
Because of this: https://github.com/lancedb/lancedb/pull/247#issuecomment-1618994287 (above), I asked Chang about this and he mentioned to me about ways to disable lint for specific lines. The entire PR is based around fixing the docs generation but because of this method overloading syntax isn't recognized by `typedoc`, we had to either disable lint for those lines or open an issue with `typedoc` themselves, which may take a long time

Reply:
But what do you think? Should I fix the docs another way, or continue with what I have in this PR?